docs/naps/7-key-binding-dispatch.md

```diff
@@ -0,0 +1,427 @@
+- `ctrl+x alt` is **invalid** because it is a key chord whose second part is a single modifier
+- `ctrl+x alt+v` is **valid** as a key chord
+- `meta meta` is **invalid** because it is a key chord comprised of only single modifier parts
+
+## Detailed Description
+
+Even with conditional activation, many key bindings may find that they share the exact same key sequence as another key binding (a direct conflict), or that their key sequence is a subset of another key binding's key sequence (an indirect conflict). This requires the establishment of a system to determine when and how key bindings should be dispatched.
+
+### Key binding properties
+
+All key binding entries contain the following information:
+- `command_id` is the unique identifier of the command that will be executed by this key binding
+- `weight` is the main determinant of key binding priority. high value means a higher priority
+- `when` is the context expression that is evaluated to determine if the key binding is active; if not provided, the key binding is always considered active
+- (autoset) `block_rule` is enabled if `command_id == ''` and disables all lower priority key bindings
+- (autoset) `negate_rule` is enabled if `command_id` is prefixed with `-` and disables all lower priority key bindings with the same sequence bound to this command
+
+```python
+from dataclasses import dataclass, field
+
+from app_model.expressions import Expr
+
+@dataclass(order=True)
+class KeyBindingEntry:
+    command_id: str = field(compare=False)
+    weight: int
+    when: Optional[Expr] = field(compare=False)
+    block_rule: bool = field(init=False)
+    negate_rule: bool = field(init=False)
+    
+    def __post_init__(self):
+        self.block_rule = self.command_id == ''
+        self.negate_rule = self.command_id.startswith('-')
+```
+
+### Direct conflicts
+
+When two key bindings share the same key sequence, they are considered to be in direct conflict. They are sorted first according to their weight, then whether they are a blocking rule, whether they are a negate rule, and otherwise, based on their insertion order. This is done in ascending order such that higher weights and blocking/negate rules are moved further down the list.
+
+Key bindings will automatically be assigned weights depending on who set them, prioritizing default ones the least and user-set ones the most:
+
+```python
+from enum import IntEnum
+
+class KeyBindingWeights(IntEnum):
+    CORE = 0
+    PLUGIN = 300
+    USER = 500
+```
+
+### Indirect conflicts
+
+When a key sequence matches a key binding and is also a sub-sequence of a key sequence used by another key binding, it is considered an indirect conflict.
+
+There are two ways indirect conflicts can exist:
+
+A. the provided key sequence is a single modifier that is a modifier in another key binding's key combination or is a modifier in the first key combination of a key binding's key chord
```

Comment:
Maybe I don't have the right kind of brain to think about key bindings, but an example for each case would help a lot!